### PR TITLE
[SPDBT-2915]  crc controlling memeber

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

- add contract (manager) and cotnroller for crc controlling member(unauth post)

### CODE REVIEW

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using MediatR;
+using Spd.Manager.Shared;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Spd.Manager.Licence;
+public abstract record CrcControllingMemberApp
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/CrcControllingMemberContract.cs:10]: no need to add Crc, just ControllingMemberApp

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{
+    public int AccessCode { get; set; } //is it int or string?
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/CrcControllingMemberContract.cs:12]: it should be a string, I am not sure request will contain accessCode, you can leave it here.

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+public record CrcControllingMemberCommandResponse : CrcControllingMemberUpsertResponse
+{
+};
+
+public record CrcControllingMemberUpsertResponse
+{
+}
+public record CrcControllingMemberUpsertRequest : CrcControllingMemberApp
+{
+    public Guid? CrcControllingMemberId { get; set; }
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/CrcControllingMemberContract.cs:59]: should be ControllingMemberAppId

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public bool HasBankruptcyHistory { get; set; }
+    public string? BankruptcyHistoryDetail { get; set; }
+
+    public bool HasMentalCondition { get; set; }
+    //public LicenceAppDocumentUpload? MentalHealthConditionDoucuments { get; set; }
+
+    public Address Address { get; set; }
+}
+
+public record CrcControllingMemberCommandResponse : CrcControllingMemberUpsertResponse
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/CrcControllingMemberContract.cs:50]: Response should contain ControllingMemberAppId

File: src/Spd.Presentation.Licensing/Controllers/CrcControllingMemberController.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        _mediator = mediator;
+    }
+
+    /// <summary>
+    /// Save New Licence Crc Controlling Member
+    /// </summary>
+    /// <param name="CrcControllingMemberUpsertRequest"></param>
+    /// <returns></returns>
+    [Route("api/business-licence/New-Crc-Controlling-Member")]
```
Comment (peggy-quartech) [inline src/Spd.Presentation.Licensing/Controllers/CrcControllingMemberController.cs:30]: api/controlling-member-crc-applications

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs (hunk 3 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Licence/CrcControllingMemberContract.cs:59]: add ControllingMemberCrcAppSubmitRequest : CrcControllingMemberApp

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+{
+    public int AccessCode { get; set; } //is it int or string?
+    public string GivenName { get; set; }
+    public string? MiddleName1 { get; set; }
+    public string? MiddleName2 { get; set; }
+    public string Surname { get; set; }
+    public bool IsTermsOfUseAccepted { get; set; }
+    public DateTime DateSigned { get; set; }
+    public DateTime DateOfBirth { get; set; }
+    public GenderCode Gender { get; set; }
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/CrcControllingMemberContract.cs:20]: GenderCode

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Manager.Licence/CrcControllingMemberContract.cs (hunk 3 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/CrcControllingMemberContract.cs:59]: removed from request

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**